src/PreCICE.jl

```diff
@@ -64,227 +85,1031 @@ function createSolverInterfaceWithCommunicator(participantName::String,
     return ans
 end
 
 
+@doc """
+
+    markActionFulfilled(action::String)
+
+Indicate preCICE that a required action has been fulfilled by a solver. 
+
+# Arguments
+ - `action::String`: Name of the action.
+
+# Notes
+
+Previous calls:
+ - The solver fulfilled the specified action.
+"""
 function markActionFulfilled(action::String)
     ccall((:precicec_markActionFulfilled, "libprecice"), Cvoid, (Ptr{Int8},), action)
 end
 
 
+@doc """
+
+    hasMesh(meshName::String)::Bool
+
+Check if the mesh with given name is used by a solver. 
+"""
 function hasMesh(meshName::String)::Bool
     ans::Integer = ccall((:precicec_hasMesh, "libprecice"), Cint, (Ptr{Int8},), meshName)
     return ans
 end
 
 
+@doc """
+
+    getMeshID(meshName::String)::Integer
+
+Return the ID belonging to the given mesh name.
+
+# Examples
+
+```julia
+meshid = getMeshID("MeshOne")
+```
+"""
 function getMeshID(meshName::String)
     ans::Integer = ccall((:precicec_getMeshID, "libprecice"), Cint, (Ptr{Int8},), meshName)
     return ans
 end
 
 
+@doc """
+
+    hasData(dataName::String, meshID::Integer)::Bool
+
+Check if the data with given name is used by a solver and mesh.
+Return true if the mesh is already used.
+
+# Arguments
+ - `dataName::String`: Name of the data.
+ - `meshID::Integer`: ID of the associated mesh.
+
+"""
 function hasData(dataName::String, meshID::Integer)::Bool
     ans::Integer = ccall((:precicec_hasData, "libprecice"), Cint, (Ptr{Int8}, Cint), dataName, meshID)
     return ans
 end
 
 
+@doc """
+
+    getDataID(dataName::String, meshID::Integer)::Integer
+
+# Arguments
+- `dataName::String`: Name of the data.
+- `meshID::Integer`: ID of the associated mesh.
+
+Return the data id belonging to the given name.
+
+The given name (`dataName`) has to be one of the names specified in the configuration file. The data ID obtained can be used to read and write data to and from the coupling mesh.
+"""
 function getDataID(dataName::String, meshID::Integer)
     id::Integer = ccall((:precicec_getDataID, "libprecice"), Cint, (Ptr{Int8}, Cint), dataName, meshID)
     return id
 end
 
 
+@doc """
+
+    setMeshVertex(meshID::Integer, position::AbstractArray{Float64})
+
+Create a mesh vertex on a coupling mesh and return its id.
+
+# Arguments
+- `meshID::Integer`: The id of the mesh to add the vertex to. 
+- `position::AbstractArray{Float64}`: An array with the coordinates of the vertex. Depending on the dimension, either [x1, x2] or [x1,x2,x3].
+
+# See also
+
+[`getDimensions`](@ref), [`setMeshVertices`](@ref)
+
+# Notes
+
+Previous calls:
+ - Count of available elements at position matches the configured dimension.
+
+# Examples
+```julia
+v1_id = setMeshVertex(mesh_id, [1,1,1])
+```
+"""
 function setMeshVertex(meshID::Integer, position::AbstractArray{Float64})
     id::Integer = ccall((:precicec_setMeshVertex, "libprecice"), Cint, (Cint, Ref{Float64}), meshID, position)
     return id
 end
 
 
+# TODO Example has wrong arguments
```

Comment:
Do you want some clarification here?

Reply:
Yes, I'm not sure what is correct here. Does the c function return something or save the vertices in the `positions::AbstractArray{Float64}`? 
And should we also pass the array by reference or return the resulting `positions`?

Reply:
The intended use of `getMeshVertices` is that the user will provide a mesh ID and several vertex IDs and the function will return the corresponding vertex coordinates. We should return the resulting `positions` and not the reference.

Reply:
Okay so I should change the function to not take positions as an input argument but instead to return the result of the c function call?

Reply:
Yes that is the appropriate usage